python/lsst/ctrl/bps/bps_core.py

```diff
@@ -118,43 +118,31 @@ def pretty_dataset_label(orig_name):
     return new_name
 
 
-def save_single_qgnode(qgnode, out_filename):
-    """Save single quantum to file
+def save_qg_subgraph(node_ids, qgraph, out_filename):
+    """Save subgraph to file
 
     Parameters
     ----------
-    qgnode : QuantumGraph Node
-        Single quantum to save
+    node_ids : `lsst.pipe.base.graph.quantumNode.NodeId` or
+               iterable of `lsst.pipe.base.graph.quantumNode.NodeId`
+        NodeIds for Quanta inside given qgraph to save
     out_filename : `str`
         Name of the output file
     """
-    os.makedirs(os.path.dirname(out_filename), exist_ok=True)
-    qgraph2 = QuantumGraph()
-    qgraph2.append(qgnode)
-    with open(out_filename, "wb") as pickle_file:
-        pickle.dump(qgraph2, pickle_file)
-
 
-def count_quantum(qgraph):
-    """Count Quantum in the QuantumGraph
+    # (disabling pylint warning because pylint bug
+    #  https://github.com/PyCQA/pylint/issues/3507)
+    if not isinstance(node_ids, Iterable):   # pylint: disable=W1116
```

Comment:
Are there any plans to move generic classes or functions like these out of daf_butler?

Reply:
We are blocked at the moment by the utils package having C++ dependencies which makes me reticent to depend even more on it (it makes it really hard to distribute butler on PyPI for example). Hopefully at some point we can split utils into a C++ utils and python utils and that would free us to migrate.

Given how reliant you are on daf_butler anyhow I don't think there's a problem using the utility functions for now.

Reply:
I just pushed the changes that include using the daf_butler piece.  Since the "nodes_ids to qnodes conversion below" didn't change that much, I'm not sure whether you were thinking differently.

Reply:
I was mainly referring to remove the two lines here and making the whole thing a bit cleaner. Your change is what I had in mind.